client_auth.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2016 Mender Software AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/mendersoftware/log"
+	"github.com/pkg/errors"
+)
+
+type AuthGetter interface {
+	Request(identity, deviceID string) (interface{}, error)
+}
+
+// source of authorization data
+type AuthDataProvider interface {
+	// returns device public key
+	Pubkey() (string, error)
+	// returns tenant token
+	Token() (string, error)
+	// returns identity data
+	IdentityData() (interface{}, error)
+}
+
+type AuthClient struct {
+	client *http.Client
```

Comment:
I might remove this later on when I will go through the rest of the code, but for now I am wondering why we need `AuthClient` to be separate entity. Won't it be more intuitive if we will have `Updater` only? We can rename it as well, but essentially we need just one thing, and the fact that this is authenticated or not is just a state of the client. What is more, let's assume we will add reporting. Will we add new `Reporter` struct and add some interface and implementation for sending reports? Or maybe we will reuse the same client to do everything (send reports, ask for update and authenticate).
The only reason I can see to split this is if (or when) we will use different protocols/methods to send reports and to fetch update.